privacy-on-beam/pbeam/mean_test.go

```diff
@@ -718,6 +718,47 @@ func TestMeanPerKeyReturnsNonNegativeFloat64(t *testing.T) {
 	}
 }
 
+// Expect non-negative results with partitions if MinValue >= 0 for float64 values.
+func TestMeanPerKeyWithPartitionsReturnsNonNegativeFloat64(t *testing.T) {
+	var triples []tripleWithFloatValue
+	for key := 0; key < 100; key++ {
+		triples = append(triples, tripleWithFloatValue{key, key, 0.01})
+	}
+	p, s, col := ptest.CreateList(triples)
+	col = beam.ParDo(s, extractIDFromTripleWithFloatValue, col)
+
+	var partitions []int
+	for p := 0; p < 200; p++ {
+		partitions = append(partitions, p)
+	}
+	partitionsCol := CreateList(s, partitions)
+
+	// Using a low ε, a high δ, and a high maxValue here to add a
+	// lot of noise while keeping partitions.
+	// ε=0.001. δ=0.999 and l0Sensitivity=1 gives a threshold of =2.
+	maxContributionsPerPartition := int64(1)
+	epsilon := 0.001
+	delta := 0.999
+	lower := 0.0
+	upper := 1e8
+
+	// ε is split by 2 for noise and for partition selection, so we use 2*ε to get a Laplace noise with ε.
```

Comment:
This comment made me realize a HUGE thing we forgot to address: We shouldn' spend budget on thresholding when we specify partitions.

Please add logic for that for all aggregations.

And please fix the comments and parameters for all tests with specified partitions.